test/e2e/wireguard_test.go

```diff
@@ -40,25 +40,22 @@ func TestWireGuard(t *testing.T) {
 		t.Fatalf("Error when setting up test: %v", err)
 	}
 	defer teardownTest(t, data)
+	skipIfMulticastEnabled(t, data)
 	skipIfEncapModeIsNot(t, data, config.TrafficEncapModeEncap)
 	for _, node := range clusterInfo.nodes {
 		skipIfMissingKernelModule(t, data, node.name, []string{"wireguard"})
 	}
 	var previousTrafficEncryptionMode string
-	var previousMulticastEnabledState bool
```

Comment:
After sync with wenying, we keep both IPSec and WireGuard test in the same way to simply disabled related encryption e2e tests when multicast is enabled. So no need to do this kind of check and restore anymore.